### PR TITLE
Linting workflow

### PR DESCRIPTION
**Pull request by Ninja243**
Added workflow for the linting of the go files that runs when a pull request is made, automatically bumping the version number in the process.

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        go-version: [1.14.x]
+    runs-on: ubuntu-18.04
+    steps:
+      # Get the code into the container
+      - name: Checkout
+        uses: actions/checkout@v2
+      # Test the code
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v2.5.2
+      - name: Test code
```
Comment (aavshr) [inline .github/workflows/pull_request.yml:20]: did this step work without setting the test environment vars from github secrets?

https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

**Comment by aavshr** [comment]
Great PR @Ninja243 . Thanks!

Can you please also change the environment variable names in the test code to be `DETA_SDK_TEST_PROJECT_KEY` and `DETA_SDK_TEST_BASE_NAME` ? As you suggested, I want the environment vars to be consistent across all sdks.

**Review by aavshr — CHANGES_REQUESTED** [review]
Great PR @Ninja243 . Thanks!

Can you please also change the environment variable names in the test code to be `DETA_SDK_TEST_PROJECT_KEY` and `DETA_SDK_TEST_BASE_NAME` ? As you suggested, I want the environment vars to be consistent across all sdks.

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+name: Lint on pull request
```
Comment (aavshr) [inline .github/workflows/pull_request.yml:1]: Sorry, forgot here too, the name should also suggest that we're running tests.

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (Ninja243) [reply .github/workflows/pull_request.yml:25]: It did not but should now!

**Review by aavshr — APPROVED** [review]

**Pull request merged**